README.adoc

```diff
@@ -82,6 +82,64 @@ Once that is finished, the wallet registry can call
 `Wallets.registerNewWallet`, which sets the new wallet as the active wallet.
 Going forward, it will receive deposits.
 
+==== Wallet Closure
+
+Wallets can close in a few ways:
+
+* The operators notify the chain that the wallet is failing a heartbeat: `Wallets.notifyWalletHeartbeatFailed`
+* Someone notifies the chain that the wallet timed out while filling a redemption: `Wallets.notifyWalletTimedOutRedemption`
+* Someone notifies the chain that the wallet is too old: `Wallets.notifyCloseableWallet`
+* Someone notifies the chain that the wallet has too few BTC remaining: `Wallets.notifyCloseableWallet`
+
+Each of the above routes into `Wallets.moveFunds`. If the wallet has no main
+UTXO, then we can start closing it immediately via
+`Wallets.beginWalletClosing`. Otherwise, we change it's state to
+`WalletState.MovingFunds`, decrease the live wallet count, and give the wallet
+a deadline to move its funds to other wallet(s). If this was the active wallet,
+then we currently have no active wallet.
+
+After `movingFundsTimeout` goes by, anyone can call
+`MovingFunds.notifyMovingFundsTimeout` which pipes into
+`Wallets.terminateWallet` followed by `ecdsaWalletRegistry.seize`.
+`terminateWallet`, in this case passes through to
+`ecdsaWalletRegistry.closeWallet` after changing the state to
+`WalletState.Terminated`.
+
+Furthermore, the `ecdsaWalletRegistry.seize` call is punishing the operators by
+`movedFundsSweepTimeoutSlashingAmount` and rewarding the notifier with a reward
+multiplier of `movedFundsSweepTimeoutNotifierRewardMultiplier` (the notifier
+gets a percentaged of the slashed stake).
+
+In order to avoid this, the wallet has to commit to which wallets they'll send
+the funds to (`MovingFunds.submitMovingFundsCommitment`), then actually send
+the funds to those wallets, and then prove that they did it
+(`MovingFunds.submitMovingFundsProof`) before the time runs out.
+
+The commitment involves submitting a list of wallet public keys that:
+
+* Aren't the source wallet 
+* Are in ascending order
+* Are Live
+
+We store the hash of the list in `movingFundsTargetWalletsCommitmentHash`.
+Off-chain we come to consensus by picking the Live wallets whose public key
+hashes are the closest to the source wallet's public key hash in terms of clock
+distance (modulus distance). This makes public key hashes in the middle of the
+range no more likely to be picked than ones near the ends. We pick a number of
+wallets equal to `min(liveWalletsCount, ceil(walletBtcBalance /
+walletMaxBtcTransfer))`, where `walletMaxBtcTransfer` is governable.
+
+In `submitMovingFundsProof`, we prove that the bitcoin transaction happened and
```

Comment:
```suggestion
In `submitMovingFundsProof`, we prove that the Bitcoin transaction happened and
```

Just to keep it aligned with other occurrences.

Reply:
🙇

Reply:
[6461fa3](https://github.com/keep-network/tbtc-v2/pull/327/commits/6461fa3de6bcf9f48a49ffbda8fe3983cc1ff9dd)